### PR TITLE
Remove unused old files

### PR DESCRIPTION
**Pull request by brianhelba**
* Remove unused JSHint files
* Remove unused and outdated Heroku files
* Remove unused Bower file
  * Girder is not and probably should not be published on Bower.
* Remove outdated files for garant dashboard
  * Even if it was up-to-date (which it isn't) these machine-specific dashboard files probably shouldn't all be published and maintained in the upstream Girder repo.

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1 +0,0 @@
-web: python -m girder --port ${PORT}
```
Comment (zachmullen) [inline Procfile]: There is a section in our deployment docs about Heroku, and it mentions these files and kind of makes it sound like they are important to deploy onto Heroku. @cpatrick might know whether these are still required.

File: cmake/garant_nightly.cmake [hunk 2]
```diff
@@ -1,38 +0,0 @@
-set(CTEST_SOURCE_DIRECTORY "$ENV{HOME}/Dashboards/girder")
```
Comment (zachmullen) [inline cmake/garant_nightly.cmake]: @jbeezley is garant nightly now defunct?

Comment (jbeezley) [inline cmake/garant_nightly.cmake]: Sadly, I haven't gotten around to setting it up again.  If I need the files, I can get them from the history.

**Comment by danlamanna** [comment]
FWIW, I'm going to be adding Beanstalk configurations to the repo soon under the directory `devops/beanstalk`, and our instructions will tell users to copy these files into their Girder checkout. I think that approach makes more sense than keeping configurations for every possible platform in our root directory.

File: Procfile (hunk 1 above)
Comment (brianhelba) [inline Procfile]: Given that the `Procfile` is so obviously out of date, I'm going to consider Heroku to now be unsupported via our own neglect, and thus mercilessly remove that section of the docs.

If anyone wants to pick up the baton and fix Heroku deployment (preferably with some sort of testing mechanism), that'd be fine, but until then, we shouldn't (IMHO) document and have cruft for a broken configuration.

Comment (brianhelba) [inline Procfile]: If we do care about Heroku deployment, we should file a bug for the issue.

Regardless, right now, the config is broken, and should be removed. If anyone wants to work on the bug, they can resurrect the old docs from Git's history.

Comment (brianhelba) [inline Procfile]: Ok, I've filed https://github.com/girder/girder/issues/2078 , in case anyone wants to ever come back to it.

**Comment by brianhelba** [comment]
@girder/developers PTAL.

**Pull request merged**